Solaris doesn't define d_type dirent member and DT_REG type.

--- a/src/source-pcap-file-directory-helper.c
+++ b/src/source-pcap-file-directory-helper.c
@@ -304,7 +304,7 @@ struct dirent * dir = NULL;
     PendingFile *file_to_add = NULL;
 
     while ((dir = readdir(pv->directory)) != NULL) {
-#ifndef OS_WIN32
+#if defined(DT_REG)
         if (dir->d_type != DT_REG) {
             continue;
         }
--- a/src/util-path.c
+++ b/src/util-path.c
@@ -199,7 +199,7 @@ *          and parent directory will ret
  */
 bool SCIsRegularDirectory(const struct dirent *const dir_entry)
 {
-#ifndef OS_WIN32
+#if !defined(OS_WIN32) && !defined (__sun)
     if ((dir_entry->d_type == DT_DIR) &&
         (strcmp(dir_entry->d_name, ".") != 0) &&
         (strcmp(dir_entry->d_name, "..") != 0)) {
@@ -217,7 +217,7 @@ * \retval True if the object is a regula
  */
 bool SCIsRegularFile(const struct dirent *const dir_entry)
 {
-#ifndef OS_WIN32
+#if defined(DT_REG)
     return dir_entry->d_type == DT_REG;
 #endif
     return false;
